src/cache.service.spec.ts

```diff
@@ -0,0 +1,218 @@
+/* tslint:disable:no-unused-variable */
+
+import { TestBed, inject } from '@angular/core/testing';
+
+import {
+  HttpTestingController,
+  HttpClientTestingModule
+} from '@angular/common/http/testing';
+
+import { APIService } from './api.service';
+import { ConfigurationService } from './configuration.service';
+import { AuthenticationService } from './authentication.service';
+import { CacheService } from './cache.service';
+import { Observable } from 'rxjs/Observable';
+
+const front_page_response = {
+      "@id": "http://fake/Plone/",
+      "@type": "Plone Site",
+      "id": "Plone",
+      "items": [
+        {
+          "@id": "http://fake/Plone/front-page",
+          "@type": "Document",
+          "description": "Congratulations! You have successfully installed Plone.",
+          "title": "Welcome to Plone"
+        }
+      ],
+      "items_total": 5,
+      "parent": {}
+    };
+
+const events_page_response = {
+      "@id": "http://fake/Plone/events",
+      "@type": "Folder",
+      "id": "Plone",
+      "items": [
+        {
+          "@id": "http://fake/Plone/events/event1",
+          "@type": "Document",
+          "description": "",
+          "title": "Event 1"
+        }
+      ],
+      "items_total": 1,
+      "parent": {}
+    };
+
+describe('CacheService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        APIService,
+        ConfigurationService,
+        AuthenticationService,
+        {
+          provide: 'CONFIGURATION', useValue: {
+          BACKEND_URL: 'http://fake/Plone',
+          CACHE_REFRESH_DELAY: 1000,
+        }
+        },
+        CacheService,
+      ]
+    });
+  });
+
+  afterEach(() => {
+    TestBed.get(CacheService).revoke.emit();
+  });
+
+  it('should cache and then get cached content', inject([CacheService], (cache) => {
```

Comment:
Instead of `inject([CacheService], (cache)`, the new style is to get it in the body just like `HttpTestingController`, i.e. 
`const http = TestBed.get(CacheService);`
In fact, if it's needed in every test case you can just get it in `beforeEach`.